### DISCUSSION TITLE
Possible modifications to string class CString

### DISCUSSION THREAD

**Discussion by SteveSpencer** [rsta2/circle#479] [Ideas]
Would there be any interest in making some changes to the CString class?

Adding:
       const char* c_str()
           Performs the same function as explicit cast to const char*.
           In my head, this is clearer than a cast, as  str.c_str() is a familiar idiom to developers who have used std::string
       void Append(const char ch)
           Would be useful when creating strings from sub-strings of other strings.
       CString& operator += (const char ch)
       CString& operator += (const char* str)
       CString& operator += (const CString& str)
           Again, would be useful building CString objects

As most of the functionality already exists, there would be minimal overhead in adding these, and existing code would still recompile cleanly.
I would be happy to submit a PR to do this.

**Comment by rsta2** [comment]
Good idea. Yes, please send a PR for these new methods.

**Reply by SteveSpencer** [reply]
Done.
